### PR TITLE
Upgraded Docker File, python modules and yarn

### PR DESCRIPTION
**Pull request by brahmaduttau**

### CODE REVIEW

File: config/settings/base.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 # flake8: noqa
 
+import logging
```
Comment (alvak1) [inline config/settings/base.py:7]: is this required?

File: config/settings/base.py [hunk 2]
```diff
@@ -646,3 +648,9 @@
             },
         },
     }
+
+if env("GDAL_LIBRARY_PATH", default=None):
+    GDAL_LIBRARY_PATH = env("GDAL_LIBRARY_PATH")
+
+if env("GEOS_LIBRARY_PATH", default=None):
```
Comment (alvak1) [inline config/settings/base.py:655]: s this env key set somewhere?

File: docker/local/django/Dockerfile [hunk 3]
```diff
@@ -1,60 +1,59 @@
-ARG BASE_IMAGE=python:3.9-alpine3.16
+ARG BASE_IMAGE=python:3.10-slim
```
Comment (alvak1) [inline docker/local/django/Dockerfile:1]: As discussed - lets explore build stages to build debian images for local and alpine for other envs

File: docker/local/django/Dockerfile [hunk 4]
```diff
@@ -1,60 +1,59 @@
...
     && pip install poetry==$POETRY_VERSION \
     && poetry config virtualenvs.create false \
-    && poetry run python -m pip install 'pyproj<3' --no-build-isolation \
-    && poetry install
+    && poetry install --no-interaction --no-ansi -vvv
 
-ARG UID=1000
-ARG GID=36
+# ARG UID=1001
+# ARG GID=36
```
Comment (alvak1) [inline docker/local/django/Dockerfile:51]: We can delete these lines if not used

File: docker/local/django/Dockerfile.alpine [hunk 5]
```diff
@@ -0,0 +1,92 @@
+ARG BASE_IMAGE=python:3.10-alpine
```
Comment (alvak1) [inline docker/local/django/Dockerfile.alpine:1]: If we are using build stages then this file might not be required

File: transit_odp/otc/dataclasses.py [hunk 6]
```diff
@@ -98,33 +102,38 @@ def empty_to_none(cls, v):
...
 
-    @validator("registration_status")
+    @field_validator("registration_status")
+    @classmethod
     def validate_registration_status(cls, v):
         if v is not None and v not in ALLOWED_REGISTRATION_STATUSES:
             raise ValueError(f"Invalid registration status: {v}")
         return v
 
+    # TODO[pydantic]: We couldn't refactor the `validator`, please replace it by `field_validator` manually.
```
Comment (alvak1) [inline transit_odp/otc/dataclasses.py:129]: Manual update of validator to field_validator required

File: .github/workflows/dev_deployment.yml [hunk 7]
```diff
@@ -2,6 +2,7 @@ on:
   push:
     branches:
     - 'dev'
+    - 'debugger'
```
Comment (alvak1) [inline .github/workflows/dev_deployment.yml:5]: Should this be removed before merging to dev?

**Review by ankitmahajan1 — APPROVED** [review]
looks good

**Review by ankitmahajan1 — APPROVED** [review]
looks good

**Pull request merged**